packages/devtools_app/lib/src/memory/memory_controller.dart

```diff
@@ -992,33 +1140,118 @@ class MPChartData {
   }
 }
 
+/// Prepare data to plot in Event Chart.
+class MPEventsChartData {
+  bool _pruning = false;
+
+  /// Signal that every addTrace will cause a prune.
+  bool get pruned => _pruning;
+
+  /// Datapoint entries for ghost trace to stop auto-scaling of Y-axis.
+  final ghosts = <Entry>[];
```

Comment:
it seems these lists all have the same # of entries which is a little bit surprising. Maybe document that and consider having 
List<List<Entry>> with enum ids for the various data point types.
As is, it looks like you would need to update a number of places in the code to add a new entry type.

Reply:
They all have the same number of entries to keep the timeline (ticks) in proper sync with the main chart as we scroll around, zoom in/out, etc.  I think a better solution, if memory becomes a problem, would be to only expand the entries for the visible ticks.

However, List<List> with enum ids for each dataset is a good idea.

Reply:
Fixed for all three charts (event, adb, and event pane).